### PR TITLE
🐛 Make cloud config secret key backward compatible

### LINKED ISSUE #1357: Cluster upgrade fails when capz is upgraded to version >= 04.9
/kind bug

**What steps did you take and what happened:**

1. Deploy capz version < 0.4.9
2. Create cluster 
3. Upgrade capz to >=0.4.9
4. Upgrade cluster created previously

cluster fails to upgrade because it cant find the secret `azure.json`

**What did you expect to happen:**
Cluster should upgrade successfully

**Anything else you would like to add:**

#955 changed cloud config secret keys from `azure.json` to `control-plane-azure.json` and `worker-node-azure.json`.

slack thread: https://kubernetes.slack.com/archives/CEX9HENG7/p1619814825276400

**Environment:**

- cluster-api-provider-azure version: 
- Kubernetes version: (use `kubectl version`): 
- OS (e.g. from `/etc/os-release`):

### PR DESCRIPTION
**Pull request by shysank**
**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:

Copies `control-plane-azure.json` to `azure.json` for backwards compatibility.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #1357

**Special notes for your reviewer**:

**TODOs**:

- [x] squashed commits
- [ ] includes documentation
- [x] adds unit tests

**Release note**:

```release-note
 Make azure.json secret key backward compatible
```

### CODE REVIEW

**Comment by nader-ziada** [comment]
/lgtm
/approve

re-running the windows test to get past the timeout

File: controllers/helpers.go [hunk 1]
```diff
@@ -188,6 +188,7 @@ func GetCloudProviderSecret(d azure.ClusterScoper, namespace, name string, owner
 	secret.Data = map[string][]byte{
 		"control-plane-azure.json": controlPlaneData,
 		"worker-node-azure.json":   workerNodeData,
+		"azure.json":               controlPlaneData,
```
Comment (CecileRobertMichon) [inline controllers/helpers.go:191]: /hold

can you please add a code comment that says this is included for backward compatibility?

**Comment by CecileRobertMichon** [comment]
/hold cancel
/lgtm

thanks @shysank

**Pull request merged**